### PR TITLE
Avoid running subsystem-benchmarks with nextest

### PR DESCRIPTION
**Pull request by AndreiEres**
Fixes https://github.com/paritytech/polkadot-sdk/issues/3704

### CODE REVIEW

**Review by ggwpez — APPROVED** [review]
(no opinion on the code but it fixes my issue)

**Comment by bkchr** [comment]
I don't get how this fix fixes the issue?

I also have the fundamental question, why do we have a benchmark registered as `test`?  Registering the benchmark with `harness = false` should be achieve the same result? 

Generally not sure why this is also not just wrapped into Criterion. I mean you have there logic for warmup etc that would all be done by criterion automatically.

**Comment by AndreiEres** [comment]
@bkchr 
Tests require `subsystem-benchmarks` feature, but they also depend on `polkadot-availability-recovery` with the same feature, so `nextest` runs them. That was the problem.

About why it's done by tests: they are a bit specific, so we need to implement them in that way without criterion.

**Comment by bkchr** [comment]
> About why it's done by tests: they are a bit specific, so we need to implement them in that way without criterion.

Okay fine, but you don't need to use criterion. If you use `harness = false` you basically have to write your main on your own, like you have done.

**Comment by AndreiEres** [comment]
> If you use `harness = false` you basically have to write your main on your own, like you have done.

Yes, but what do you mean?

File: polkadot/node/network/availability-recovery/Cargo.toml [hunk 1]
```diff
@@ -51,3 +51,4 @@ required-features = ["subsystem-benchmarks"]
 
 [features]
 subsystem-benchmarks = []
+subsystem-benchmarks-helpers = []
```
Comment (sandreim) [inline polkadot/node/network/availability-recovery/Cargo.toml:54]: `subsystem-benchmarks` feature seems to no longer be used at all.

**Review by sandreim — COMMENTED** [review]
It's kind of odd to have 2 features in use now. I also don't really get why this fixes the issue.

File: polkadot/node/network/availability-recovery/Cargo.toml (hunk 1 above)
Comment (AndreiEres) [reply polkadot/node/network/availability-recovery/Cargo.toml:54]: It's in use. We require that feature to run regression tests. That way we don't mix normal and regression tests.

**Comment by sandreim** [comment]
Can't the test just be skipped if feature not defined ?

**Comment by bkchr** [comment]
> Yes, but what do you mean?

I mean this: https://github.com/paritytech/polkadot-sdk/pull/3741

**Comment by bkchr** [comment]
Closing in favor of: https://github.com/paritytech/polkadot-sdk/pull/3741

**Pull request closed without merging**